src/register/sr.rs

```diff
@@ -4,6 +4,7 @@ use crate::asm;
 
 /// Status Register
 #[derive(Clone, Copy, Debug)]
+#[repr(C)]
```

Comment:
If we want `Sr` to have the exact layout as `u16`, then we should use `repr(transparent)` instead.